cformers/interface.py

```diff
@@ -108,57 +99,36 @@ def generate(self,
-                   "-m", self.model_save_path,
-                   "--prompt", prompt,
-                   "--seed", str(seed),
-                   "--threads", str(n_threads),
-                   "--n_predict", str(num_tokens_to_generate),
-                   "--top_k", str(top_k),
-                   "--top_p", str(top_p),
-                   "--temp", str(temperature),
-                   "--repeat_last_n", str(repeat_last_n),
-                   "--repeat_penalty", str(repeat_penalty)]
-        print(" ".join(command))
-
-        process = Popen(command, stdout=PIPE, stderr=PIPE)
-        all_stdout_so_far = ""
-        for c in iter(lambda: process.stdout.read(1), b""):
-            if print_streaming_output:
-                sys.stdout.buffer.write(c)
-            all_stdout_so_far += c.decode('utf-8')
-
-            # Check if the line is empty or matches the end marker
-            if '<END|>' in all_stdout_so_far:
-                if print_streaming_output:
-                    print('\n------------------\n')
+        process = Popen(["./cpp/main", self.cpp_model_name,
+                         "-m", self.cpp_model_name,
+                         "--prompt", prompt,
+                         "--seed", str(seed),
+                         "--threads", str(n_threads),
+                         "--n_predict", str(num_tokens_to_generate),
+                         "--top_k", str(top_k),
+                         "--top_p", str(top_p),
+                         "--temp", str(temperature),
+                         "--repeat_last_n", str(repeat_last_n),
+                         "--repeat_penalty", str(repeat_penalty)],
+                        stdout=PIPE, stderr=PIPE)
+
+        while True:
+            # Wait for data to be available on the process's standard output stream
+            select.select([process.stdout.fileno()], [], [])
+
+            # Read available data from the stream and print it to the console
+            output = process.stdout.readline().decode('utf-8')
+            if output == '' and process.poll() is not None:
                 break
-
-            # # Also check for errors
-            # err = process.stderr.readline().decode('utf-8').strip()
-            # if err:
-            #     raise Exception(err)
-
-        # print('\n' + '-'*30, all_stdout_so_far, '-'*30 + '\n')
-        # return all_stdout_so_far
-        token_line = re.findall(r'<\|BEGIN\>(.*?)<END\|>', all_stdout_so_far, re.DOTALL)[0]
-
-        # Convert the token_line to a list of integers
-        all_tokens = [int(x) for x in token_line.split()]
-
-        # Decode the tokens
-        decoded_tokens = self.tokenizer.decode(all_tokens)
+            print(output.strip())
 
         # Wait for the process to finish and return its exit code
-        return {"success": process.wait(),
-                "token_ids": all_tokens,
-                "token_str": decoded_tokens}
-        
+        return process.wait()
```

Comment:
Could you also share cpp files for gpt2 quantization and changes made to main.cpp?

Reply:
Let me know if you need any help regarding it.

Reply:
Actually I used the gptj script only